runtime/src/lib.rs

```diff
@@ -1040,13 +1111,54 @@ impl_runtime_apis! {
                     weights: stats.weights,
                 },
                 params: ModuleParams {
-                    name: params.name,
-                    address: params.address,
+                    name: String::from_utf8(params.name).expect("Name bytes should be valid utf8"),
+                    address: String::from_utf8(params.address).expect("Name bytes should be valid utf8"),
                     delegation_fee: params.delegation_fee,
                     controller: params.controller,
                 }
             }
         }
+
+        fn get_key_info(key: AccountId) -> KeyInfo {
```

Comment:
This is also really inefficient. You should iterate through `StakeTo` with `iter_prefix`, to avoid iterating for modules that the key doesn't have connections to.

Currently, every time someone calls this, it will iterate through every subnet and module.

Reply:
Even better if you figure out a way to only iterate through subnets the key is present.